Plugins/ASEP/Get-ASEP.ps1

```diff
@@ -0,0 +1,176 @@
+    -accepteula    Does not prompt for EULA acceptance
+
+    This plugin opts for collection of all ASEPs available including:
+
+    Boot execute, codecs, appinit DLLs, Explorer addons, sidebar gadgets, image
+    highjacks, Internet Explorer addons, known DLLs, logon startups, WMI entries,
+    Winsock protocol and network providers, Office addins, printer monitor DLLs,
+    LSA security providers, Autostart services and non-disabled drivers, scheduled
+    tasks, winlogon entries
+
+    Note: If you are interested in running a different version of this command, you 
+    can simply edit the Invoke-Command entry below and adjust as needed.
+
+    Note: If you are using custom directory paths, you will need to edit the EXE 
+    checks in the script so that it does not error and exit. Best practice, use the
+    default paths provided by Power-Response.
+
+.EXAMPLE
+    Get-ASEP.ps1 -ComputerName Test-PC
+
+    PowerResponse Execution
+    Set ComputerName Test-PC
+    Run
+
+.NOTES
+    Author: 5yn@x
+    Date Created: 12/29/2018
+    Twitter: @5ynax
+    
+    Last Modified By:
+    Last Modified Date:
+    Twitter:
+  
+#>
+
+param (
+
+    [Parameter(Mandatory=$true,Position=0)]
+    [string[]]$ComputerName
+
+    )
+
+process{
+
+    #Get the root Power-Response directory - Assumes that the plugin is located in the plugins/ASEP directory
+    $PRRoot = (Get-Item $PSScriptRoot).parent.parent.FullName
+
+    #Autorunsc executable locations
+    $Autorunsc64 = "$PRRoot\Bin\autorunsc64.exe"
+    $Autorunsc32 = "$PRRoot\Bin\autorunsc.exe"
+
+    #Verify binaries exist in Bin
+    $64bitTestPath = Get-Item -Path $Autorunsc64 -ErrorAction SilentlyContinue
+    $32bitTestPath = Get-Item -Path $Autorunsc32 -ErrorAction SilentlyContinue
+
+    if (-not $64bitTestPath) {
+
+        Write-Error "Autorunsc64.exe not detected in Bin. Place 64bit executable in Bin directory and try again."
+        Exit 
+
+    } elseif (-not $32bitTestPath) {
+
+        Write-Error "Autorunsc.exe not detected in Bin. Place 32bit executable in Bin directory and try again."
+        Exit
+
+    }
+
+    #Loop through machines in $ComputerName to obtain data for each machine (if multiple machines are specified)
+    foreach ($Computer in $ComputerName) {
+
+    #Handle $ComputerName defined as localhost and not break the Plugin
+    if ($Computer -match "Localhost") {
+
+        $Computer = $ENV:ComputerName
+    }
+    
+    #Verify machine is online and ready for data collection
+    if (-not (Test-Connection -ComputerName $Computer -Count 1 -Quiet)) {
+        
+        Write-Error ("{0} appears to be offline. Cannot gather ASEP data." -f $Computer)
+        Continue
+    }
+
+    #Determine system architecture and select proper Autorunsc executable
+    try {
+        $Architecture = (Get-WmiObject -ComputerName $Computer -Class Win32_OperatingSystem -Property OSArchitecture -ErrorAction SilentlyContinue).OSArchitecture
```

Comment:
Typically try-catch blocks and `-ErrorAction SilentlyContinue` aren't paired together since the former looks for terminating errors and the latter suppresses terminating errors.

You could go one of two ways with this:
1. Terminating errors: keep your try-catch block switch the WMI call to `-ErrorAction Stop` to force a terminating error you can catch
2. Non-Terminating errors: keep your `-ErrorAction SilentlyContinue` and lean fully on your following if check and can remove the try-catch

Reply:
Similar idea through the next couple sections :)

Reply:
Changes made.